internal/db/collection_define.go

```diff
@@ -148,12 +152,27 @@ func (db *DB) patchCollection(
 	if err != nil {
 		return err
 	}
+	newDefinitions := make([]client.CollectionDefinition, len(existingCols))
+	updatedColsByID := make(map[uint32]struct{})
+	for i, col := range existingCols {
+		newDefinitions[i].Schema = col.Schema()
+		newDefinitions[i].Description = newColsByID[col.ID()]
+		updatedColsByID[col.ID()] = struct{}{}
+	}
+	// append new cols
+	for id, col := range newColsByID {
+		if _, ok := updatedColsByID[id]; ok {
+			continue
+		}
+		newDefinitions = append(newDefinitions, client.CollectionDefinition{Description: col})
```

Comment:
suggestion: Seems like there is no patch collection test that has the need to append new collection definitions, might be worth adding tests for if easy to do so.

Reply:
There should be as we have a test that tries to add a collection via patch. I'll investigate.

Reply:
I can confirm that this is hit with intergration tests 👍